solana/programs/example-native-token-transfers/src/instructions/transfer.rs

```diff
@@ -120,6 +120,15 @@ pub fn transfer_burn(ctx: Context<TransferBurn>, args: TransferArgs) -> Result<(
     );
 
     let accs = ctx.accounts;
+
+    // TODO If functionality is added to disable transceivers, this check should be modified to
+    // ensure that, if all transceivers have been disabled, that an error is returned. Otherwise it
+    // may be possible to permanently lose funds.
+    // No transceivers have been registered yet
+    if accs.common.config.next_transceiver_id == 0 {
```

Comment:
Can this check be put in the common account context, maybe in the `NotPausedConfig`:
```rust
#[derive(Accounts)]
pub struct NotPausedConfig<'info> {
    #[account(
        constraint = !config.paused @ crate::error::NTTError::Paused,
        constraint = config.next_transceiver_id != 0 @ NTTError::NoRegisteredTransceivers
    )]
    config: Account<'info, Config>,
}
```

This would avoid having to put this check in each transfer instruction.

Reply:
If it is undesirable to do this check for everything that uses `NotPausedConfig` btw, adding a constraint over `common` works, too:
```rust
    #[account(
        constraint = config.next_transceiver_id != 0 @ NTTError::NoRegisteredTransceivers
    )]
    pub config: NotPausedConfig<'info>,
```

Reply:
Thanks for the review!
Yeah my intention is just to guard the Transfer functions so adding that constraint in the transfer context makes sense. I'll modify the code.